File: DataStructureGo/stack.go
package main

type Stack struct {
	array []interface{}
}

func MakeStack() Stack {
	return Stack{
		array: make([]interface{}, 0),
	}
}

func (s *Stack) isEmpty() bool {
	if s.array == nil {
		return true
	}
	return len(s.array) == 0
}

func (s *Stack) push(element interface{}) {
	s.array = append(s.array, element)
}

func (s *Stack) pop() interface{} {
	length := len(s.array)
	if length == 0 {
		return nil
	}
	ret := s.array[length-1]
	s.array = s.array[:length-1]
	return ret
}

func (s *Stack) peek() interface{} {
	length := len(s.array)
	if length == 0 {
		return nil
	}
	return s.array[length-1]
}

File: DataStructureCSharp/DataStructure/Stack.cs
public class Stack<T> {

    private List<T> list { get; }

    public Stack() {
        list = new List<T>();
    }

    public bool IsEmpty {
        get {
            return list.Count == 0;
        }
    }

    public void Push(T item) {
        list.Add(item);
    }

    public T? Pop() {
        if (IsEmpty) {
            return default;
        }
        var last = list[list.Count - 1];
        list.RemoveAt(list.Count - 1);
        return last;
    }

    public T? Peek() {
        if (IsEmpty) {
            return default;
        }
        return list[list.Count - 1];
    }

    public override string ToString() {
        return $"Stack: [{String.Join(", ", list)}]";
    }

}
